item-management/bulk-update-url-field-by-vault.py

```diff
@@ -0,0 +1,335 @@
+import subprocess
+import json
+import csv
+import threading
+import concurrent.futures
+import sys
+import itertools
+import time
+from datetime import datetime
+import os
+from tqdm import tqdm
+
+# Script to update website fields for all items in a 1Password vault.
+# Allows selecting a vault, setting a new URL, retrying if needed, or reverting changes.
+
+# ANSI color codes for console output.
+CYAN = "\033[96m"
+YELLOW = "\033[93m"
+RED = "\033[91m"
+RESET = "\033[0m"
+
+# Displays a table in a styled ASCII box with column separators.
+def show_table_in_box(data):
+    # Define column headers.
+    headers = ["ItemID", "ItemTitle", "OldWebsite", "NewWebsite"]
+
+    # Calculate maximum length for each column.
+    max_lengths = {
+        "ItemID": max(max(len(item["ItemID"]) for item in data), len("ItemID")),
+        "ItemTitle": max(max(len(item["ItemTitle"]) for item in data), len("ItemTitle")),
+        "OldWebsite": max(max(len(item["OldWebsite"]) for item in data), len("OldWebsite")),
+        "NewWebsite": max(max(len(item["NewWebsite"]) for item in data), len("NewWebsite"))
+    }
+
+    # Add padding for readability.
+    max_lengths = {k: v + 2 for k, v in max_lengths.items()}
+
+    # Build the box borders.
+    top_border = "┌" + "─" * max_lengths["ItemID"] + "┬" + "─" * max_lengths["ItemTitle"] + "┬" + "─" * max_lengths["OldWebsite"] + "┬" + "─" * max_lengths["NewWebsite"] + "┐"
+    header_row = (
+        "│ " + headers[0].ljust(max_lengths["ItemID"] - 2) + " │ " +
+        headers[1].ljust(max_lengths["ItemTitle"] - 2) + " │ " +
+        headers[2].ljust(max_lengths["OldWebsite"] - 2) + " │ " +
+        headers[3].ljust(max_lengths["NewWebsite"] - 2) + " │"
+    )
+    mid_border = "├" + "─" * max_lengths["ItemID"] + "┼" + "─" * max_lengths["ItemTitle"] + "┼" + "─" * max_lengths["OldWebsite"] + "┼" + "─" * max_lengths["NewWebsite"] + "┤"
+    bottom_border = "└" + "─" * max_lengths["ItemID"] + "┴" + "─" * max_lengths["ItemTitle"] + "┴" + "─" * max_lengths["OldWebsite"] + "┴" + "─" * max_lengths["NewWebsite"] + "┘"
+
+    # Output the box.
+    print(f"{CYAN}{top_border}{RESET}")
+    print(f"{CYAN}{header_row}{RESET}")
+    print(f"{CYAN}{mid_border}{RESET}")
+    for item in data:
+        item_id = item["ItemID"].ljust(max_lengths["ItemID"] - 2)
+        item_title = item["ItemTitle"].ljust(max_lengths["ItemTitle"] - 2)
+        old_website = item["OldWebsite"].ljust(max_lengths["OldWebsite"] - 2)
+        new_website = item["NewWebsite"].ljust(max_lengths["NewWebsite"] - 2)
+        print(f"│ {item_id} │ {item_title} │ {old_website} │ {new_website} │")
+    print(f"{CYAN}{bottom_border}{RESET}")
+
+# Executes a 1Password CLI command and returns the output.
+def run_op_command(args):
```

Comment:
can we type annotate the argument here? I think we should try to do that wherever possible as a general practice (we haven't traditionally done that in Python, but especially for these more generic functions that will be re-used and which customers may rely on if they customize the script, it can help make things a bit clearer and help the IDE with code completion and suggestions, etc). 
```suggestion
def run_op_command(args: List[str]):
```

Reply:
Updated with the recommendation with commit [b23d6b6](https://github.com/1Password/solutions/pull/91/commits/b23d6b61961ffbfc7d447c4b05792b71a315e195)